File: client/src/app/shared/models/order.ts
import { Address } from './address';

export type OrderItem = {
  productId: number;
  productName: string;
  pictureUrl: string;
  price: number;
  quantity: number;
};

export type Order = {
  id: number;
  buyerEmail: string;
  orderDate: string;
  shippingAddress: Omit<Address, 'address1' | 'address2'> & { street: string };
  deliveryMethod: string;
  subtotal: number;
  deliveryCost: number;
  discount: number;
  total: number;
  status: string;
  paymentMethod: string;
  paymentStatus: string;
  orderItems: OrderItem[];
};

export type CreateOrderRequest = {
  deliveryMethodId: number;
  shipToAddress: {
    firstName: string;
    lastName: string;
    street: string;
    city: string;
    state: string;
    zipCode: string;
    country: string;
  };
  paymentMethod: string;
};

export type DeliveryMethod = {
  id: number;
  shortName: string;
  deliveryTimeDays: number;
  price: number;
};

File: Core/Entities/Order.cs
using Core.Enums;

namespace Core.Entities;

public class Order : BaseEntity
{
    public int UserId { get; set; }
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public required string BuyerEmail { get; set; }
    public required Address ShipToAddress { get; set; }
    public int DeliveryMethodId { get; set; }
    public DeliveryMethod? DeliveryMethod { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; } = [];
    public decimal Subtotal { get; set; }
    public decimal DeliveryCost { get; set; }
    public decimal Discount { get; set; }
    public decimal Total { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public string? PaymentIntentId { get; set; }
    public int? CouponId { get; set; }
    public Coupon? Coupon { get; set; }
}
